lib/Sema/ConstraintSystem.cpp

```diff
@@ -2620,10 +2619,47 @@ ConstraintSystem::getTypeOfMemberReference(
     auto *functionType = fullFunctionType->getResult()->getAs<FunctionType>();
     functionType = unwrapPropertyWrapperParameterTypes(*this, funcDecl, functionRefKind,
                                                        functionType, locator);
+    auto &ctx = DC->getASTContext();
+    auto *parentModule = useDC->getParentModule();
+    bool baseSendable = isSendableType(parentModule, baseOpenedTy);
+    bool inferredSendable =
+        ctx.LangOpts.hasFeature(Feature::InferSendableMethods);
+
+    if (inferredSendable) {
+      auto sendableProtocol = parentModule->getASTContext().getProtocol(
+          KnownProtocolKind::Sendable);
+      auto baseConformance = TypeChecker::conformsToProtocol(
+          baseOpenedTy, sendableProtocol, parentModule);
+
+      if (baseSendable) {
+        // Add @Sendable to functions without conditional conformances
+        if (baseConformance.getConditionalRequirements().empty())
+          functionType = functionType->withExtInfo(functionType->getExtInfo().withConcurrent())->getAs<FunctionType>();
+
+        // Handle Conditional Conformances
+        for (auto req : baseConformance.getConditionalRequirements()) {
+          if (!TypeChecker::conformsToProtocol(req.getFirstType(),
+                                               sendableProtocol, parentModule)
+                   .isInvalid())
+            functionType =
+                functionType
+                    ->withExtInfo(functionType->getExtInfo().withConcurrent())
+                    ->getAs<FunctionType>();
+        }
+      }
+    }
+
     // FIXME: Verify ExtInfo state is correct, not working by accident.
     FunctionType::ExtInfo info;
     openedType =
         FunctionType::get(fullFunctionType->getParams(), functionType, info);
+
+    // Add @Sendable to openedType if possible
+    if (inferredSendable && baseSendable) {
+      auto origFnType = openedType->castTo<FunctionType>();
+      openedType =
+          origFnType->withExtInfo(origFnType->getExtInfo().withConcurrent());
```

Comment:
Is the purpose of this to add `@Sendable` to the outer function type? Can we just do that unconditionally a few lines above, right here?

```
    FunctionType::ExtInfo info;
    openedType =
        FunctionType::get(fullFunctionType->getParams(), 
```

My understanding is the outer function type is always `@Sendable`. Please correct me if I'm wrong!

Reply:
The outer function type might not always be sendable for something like 

```
S<NonSendableClass>.f 

. . .

struct S<T> {
  func f() { }
}

```
where  `S<NonSendableClass>` is the outer function but this is ok because the function call is unapplied

Reply:
Updated to unconditionally mark `@Sendable`